### ISSUE TITLE
React Authenticator Throws Error After Clicking Sign Up Tab

### ISSUE THREAD

**Issue by ErikCH** [aws-amplify/amplify-ui#1623]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

Create React App and Next

### Please describe your bug.

This bug occurs only after taking a few certain steps.

After these steps are taken clicking the 'Create Account' button will throw an error in the console and the user will not be able to continue on. These errors include:

Uncaught TypeError: Cannot read properties of undefined (reading 'length')
The above error occurred in the <n> component:

### What's the expected behaviour?

No errors should be shown

### Help us reproduce the bug!

1. Using the Authenticator log into an account. (Tested with email)
2. After logging in, refresh the browser page (make sure after logging in you see a logout button).
3. Click the log out button.
4. Click the 'Create Account' tab.

Error will be thrown. (This only occurs if you refresh the page after logging in, when you see the log out button)

### Code Snippet

```typescript

function App() {
  return (
    <div className="App">
      <Authenticator>
        {({ signOut }) => <button onClick={signOut}> Sign Out</button>}
      </Authenticator>
    </div>
  );
}

export default App;


```

### Additional information and screenshots

<img width="976" alt="image" src="https://user-images.githubusercontent.com/65630/161814027-60041d40-38d3-4020-ada1-15139b6278c0.png">

<img width="1324" alt="image" src="https://user-images.githubusercontent.com/65630/161814063-d3a0f16d-4a5e-434c-a652-2daa09cafd63.png">
<img width="1168" alt="image" src="https://user-images.githubusercontent.com/65630/161814106-2f54f924-0032-487d-8791-62bc1d0c1edb.png">

**Comment by Jesmaster** [comment]
Hi guys, I'm able to replicate this issue, here are the steps:

- Log in
- Refresh the browser
- Log out
- Attempt to log in again, it will fail with the js error in the console.

**Comment by ianzone** [comment]
It happens to me when sign out and re-sign in. 

Unhandled Runtime Error
TypeError: Cannot read properties of undefined (reading '0')

Call Stack
formValues
node_modules/@aws-amplify/ui/dist/esm/machines/authenticator/actions.js (1:2001)
eval
node_modules/xstate/es/utils.js (415:0)
Array.reduce
<anonymous>
updateContext
node_modules/xstate/es/utils.js (397:0)
resolveActions
node_modules/xstate/es/actions.js (420:59)
StateNode.resolveTransition
node_modules/xstate/es/StateNode.js (887:34)
StateNode.transition
node_modules/xstate/es/StateNode.js (828:0)
eval
node_modules/xstate/es/interpreter.js (712:0)
provide
node_modules/xstate/es/serviceScope.js (8:0)
Interpreter.nextState
node_modules/xstate/es/interpreter.js (711:27)
eval
node_modules/xstate/es/interpreter.js (105:0)
Scheduler.process
node_modules/xstate/es/scheduler.js (65:0)
Scheduler.schedule
node_modules/xstate/es/scheduler.js (44:0)
Interpreter.send
node_modules/xstate/es/interpreter.js (101:0)
Interpreter.sendTo
node_modules/xstate/es/interpreter.js (133:0)
Interpreter.exec
node_modules/xstate/es/interpreter.js (799:0)
Interpreter.execute
node_modules/xstate/es/interpreter.js (201:0)
Interpreter.update
node_modules/xstate/es/interpreter.js (227:0)
eval
node_modules/xstate/es/interpreter.js (107:0)
Scheduler.process
node_modules/xstate/es/scheduler.js (65:0)
Scheduler.schedule
node_modules/xstate/es/scheduler.js (44:0)
Interpreter.send
node_modules/xstate/es/interpreter.js (101:0)
eval
node_modules/@aws-amplify/ui/dist/esm/helpers/authenticator/facade.js (1:184)
eval
node_modules/@aws-amplify/ui-react/dist/esm/components/Authenticator/hooks/useFormHandlers/useFormHandlers.js (1:474)
HTMLUnknownElement.callCallback
node_modules/react-dom/cjs/react-dom.development.js (4157:0)
Object.invokeGuardedCallbackDev
node_modules/react-dom/cjs/react-dom.development.js (4206:0)
invokeGuardedCallback
node_modules/react-dom/cjs/react-dom.development.js (4270:0)
invokeGuardedCallbackAndCatchFirstError
node_modules/react-dom/cjs/react-dom.development.js (4284:0)
executeDispatch
node_modules/react-dom/cjs/react-dom.development.js (9011:0)
processDispatchQueueItemsInOrder
node_modules/react-dom/cjs/react-dom.development.js (9043:0)
processDispatchQueue
node_modules/react-dom/cjs/react-dom.development.js (9056:0)
dispatchEventsForPlugins
node_modules/react-dom/cjs/react-dom.development.js (9067:0)
eval
node_modules/react-dom/cjs/react-dom.development.js (9258:0)
batchedUpdates$1
node_modules/react-dom/cjs/react-dom.development.js (25979:0)
batchedUpdates
node_modules/react-dom/cjs/react-dom.development.js (3984:0)
dispatchEventForPluginEventSystem
node_modules/react-dom/cjs/react-dom.development.js (9257:0)
dispatchEvent
node_modules/react-dom/cjs/react-dom.development.js (6435:0)
dispatchDiscreteEvent
node_modules/react-dom/cjs/react-dom.development.js (6410:0)

**Comment by jussiniinikoski** [comment]
#1659 Same problem?

**Comment by ianzone** [comment]
> #1659 Same problem?

yes for me.

**Comment by wlee221** [comment]
Hi all, we just merged and published the fix for this issue to the following verisons:

```
@aws-amplify/ui-angular@2.3.10
@aws-amplify/ui-react@2.15.3
@aws-amplify/ui-vue@2.3.10
``` 

Please upgrade your dependency and let us know if the problem persists, thanks! 🙏
